Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/stitch_borders.h

```diff
@@ -84,54 +85,147 @@ struct Less_for_halfedge
   const VertexPointMap& vpmap;
 };
 
-template <typename PM, typename OutputIterator, typename LessHedge, typename VertexPointMap>
+template<typename Halfedge,
+         typename Border_halfedge_map,  
+         typename Halfedge_pair,        
+         typename Manifold_halfedge_pair,
+         typename VPMAP,
+         typename Mesh>
+void fill_pairs(const Halfedge& he,
+                Border_halfedge_map& border_halfedge_map,
+                Halfedge_pair& halfedge_pairs,
+                Manifold_halfedge_pair& manifold_halfedge_pairs,
+                VPMAP vpmap,
+                const Mesh& pmesh)
+{
+  typename Border_halfedge_map::iterator set_it;
+  bool insertion_ok;
+  CGAL::cpp11::tie(set_it, insertion_ok)
+      = border_halfedge_map.insert(std::make_pair(he,std::make_pair(1,0)));
+  
+  if ( !insertion_ok ){ // we found already a halfedge with the points
+    ++set_it->second.first; // increase the multiplicity
+    if(set_it->second.first == 2)
+    {
+      if ( get(vpmap, source(he,pmesh))==get(vpmap, target(set_it->first,pmesh)) &&
+           get(vpmap, target(he,pmesh))==get(vpmap, source(set_it->first,pmesh)) )
+      {
+        set_it->second.second = halfedge_pairs.size(); // set the id of the pair in the vector
+        halfedge_pairs.push_back( std::make_pair(set_it->first, he) );
+        manifold_halfedge_pairs.push_back(true);
+      }
+    }
+    else
+      if ( set_it->second.first > 2 )
+      {
+        manifold_halfedge_pairs[ set_it->second.second ] = false;
+      }
+  }
+}
+
+template<typename Mesh,
+         typename CCMap,
+         typename FIMap>
+std::size_t num_component_wrapper(const Mesh& pmesh, 
+                      CCMap cc, 
+                      FIMap fim)
+{
+  return CGAL::Polygon_mesh_processing::connected_components(pmesh, cc, 
+                                                             CGAL::Polygon_mesh_processing::parameters::face_index_map(fim));
+}
+
+//specialization if there is no FIMap
+template<typename Mesh,
+         typename CCMap>
+std::size_t num_component_wrapper(const Mesh&, 
+                      CCMap, 
+                      bool)
+{
+  return 0;
```

Comment:
8/ Use `boost::graph_has_property<TriangleMesh, face_index_t>()` to check if a default property exist and if not using a dynamic one that you initialize and pass to cc.

Reply:
Testing it with a default Polyhedron should not work with the current code or I missed something.

Reply:
See https://gist.github.com/sloriot/315459c9339d3a69f4740d15e1172949 that was using a previous version of dynamic pmaps